### PR TITLE
feat: support lookup names in RPC addrs

### PR DESCRIPTION
**Pull request by b5**
support addrs like `grpc://example:4444`, which is required for DevOps infrastructure that provides hostname resolution as part of service discovery.

This patch uses the heuristic of being able to resolve to disambiguate between `grpc://foo` the lookup address and `grpc://foo` a path to a unix domain socket file location. In practice this shouldn't be too big a deal, but might get tricky if folks start naming UDS paths after domains 😵 .

I need this to make docker compose examples work, which is a good indication this'll be needed by all modern DevOps deployments

### CODE REVIEW

File: iroh-rpc-types/src/macros.rs [hunk 1]
```diff
@@ -19,6 +19,11 @@ macro_rules! proxy_serve {
                         Ok(())
                     }
 
+                    #[cfg(feature = "grpc")]
+                    $crate::Addr::GrpcHttp2Lookup(name) => {
+                        anyhow::bail!("cannot serve on lookup address: {}", name);
```
Comment (Arqu) [inline iroh-rpc-types/src/macros.rs:24]: Aren't we moving away from anyhow?

**Review by Arqu — APPROVED** [review]
LGTM, just a nit on the anyhow but that can be done in a separate PR once we start.

File: iroh-rpc-types/src/addr.rs [hunk 2]
```diff
@@ -114,6 +127,18 @@ mod tests {
         assert_eq!(addr.to_string(), "grpc://198.168.2.1:1234");
     }
 
+    // TODO(b5): only running this on unix b/c windows doesn't have localhost
+    // enabled by default
```
Comment (flub) [inline iroh-rpc-types/src/addr.rs:131]: you could start by doing the localhost lookup locally and exit the test early if it doesn't resolve.  that way the test can run everywhere and will do something useful if localhost exists.

File: iroh-rpc-types/src/macros.rs (hunk 1 above)
Comment (flub) [reply iroh-rpc-types/src/macros.rs:24]: The conclusion was yes we are, but not yet right now - only sometime before 1.0.

**Review by flub — APPROVED** [review]

**Pull request merged**